packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -778,6 +778,30 @@ let getSingleUseLegacyLogCallback = (logger: ITelemetryLoggerExt, type: string)
 	};
 };
 
+/**
+ * Load the stores from a snapshot and returns the runtime.
+ * @param params - An object housing the runtime properties:
+ * - context - Context of the container.
+ * - registryEntries - Mapping from data store types to their corresponding factories.
+ * - existing - Pass 'true' if loading from an existing snapshot.
+ * - runtimeOptions - Additional options to be passed to the runtime
+ * - containerScope - runtime services provided with context
+ * - provideEntryPoint - Promise that resolves to an object which will act as entryPoint for the Container.
+ * This object should provide all the functionality that the Container is expected to provide to the loader layer.
+ * @legacy
+ * @alpha
+ */
+export async function loadContainerRuntime(params: {
+	context: IContainerContext;
+	registryEntries: NamedFluidDataStoreRegistryEntries;
+	existing: boolean;
+	runtimeOptions?: IContainerRuntimeOptions;
+	containerScope?: FluidObject;
+	provideEntryPoint: (containerRuntime: IContainerRuntime) => Promise<FluidObject>;
+}): Promise<IContainerRuntime & IRuntime> {
```

Comment:
`IContainerRuntime` vs. `IRuntime` (and `IContainerRuntimeBase`?), has been a continuous point of confusion across the team (e.g. whether it provides loader-facing or runtime-facing API surface, which of the interfaces various new functionality should go on).

Maybe this is a good opportunity to have the free function return something with more clarity?  Could also be an opportunity to make some changes that would otherwise be breaking if made on ContainerRuntime.loadRuntime() directly (e.g. remove deprecated members, convert any->unknown, maybe other stuff I'm not thinking of)?

Reply:
It is required this function return and IRuntime, as it is used in consumers RuntimeFactories, which are used to bind the loader and runtime via that interface.

So far IContainerRuntime  looks ok, and seem to expose the things previous uses of the ContainerRuntime expect. 

IContainerRuntimeBase is in worse shape, but a bit unrelated here, as they mainly cast to IContainerRuntime  to get what they need. I think there is need to further refine these interfaces, IContainerRuntime is for the container author, and has more control, and IContainerRuntimeBase  base is for the datastore author, who has less control. A number of the case where consumers cast to IContainerRuntime  are probably legitimate cases where IContainerRuntimeBase  should expose the functionaly as well.

all that said, none of this really has to do with hiding the class, which is a big pain point and will block work i have planned this semester, so i think should be tackle separately.

Reply:
Maybe a followup then - suppose some customer makes a good case that they need some method/property from the class that is currently on neither `IRuntime` nor `IContainerRuntime`.  Where would that method/property go? 
 Would this return value become something like `IContainerRuntime & IRuntime & IContainerRuntimeLegacyClassExtras`?

Understood maybe none of this matters to your current project, but since we're talking about adding very core functionality here I think it'd be irresponsible to not at least consider the long-term view of it.

BTW, if you have a good understanding of those interfaces' respective roles it would be great if you could beef up their docs - currently their TSDocs don't reveal anything about this.

Reply:
I don't expect to many cases like that, as i think most the problems will be around mixins. For anything we do find, we'll need to handle it case by case, but i'm open to adding functionally to IContainerRuntime, as it's really just ignored, and the ContainerRuntime class has been evolved independently, so i don't think i necessarily need to relitigate every exposure there just to remove the class exposure. We also have some know cruft in this area, which you can see by looking at the interface IContainerRuntimeWithResolveHandle_Deprecated. I believe if we get off the class all this gets easier to deal with, as its easy to add new functions, and new interfaces that compose internal implementations to support both new and old patterns rather than being stuck to a rigid monolithic implementation that is the ContainerRuntime class.